pkg/module/service.go

```diff
@@ -0,0 +1,58 @@
+package module
+
+import (
+	"context"
+	"github.com/odpf/entropy/domain"
+	"github.com/odpf/entropy/store"
+)
+
+type ServiceInterface interface {
+	Sync(ctx context.Context, urn string) error
+}
+
+type Service struct {
+	resourceRepository store.ResourceRepository
+	moduleRepository   store.ModuleRepository
+}
+
+func NewService(resourceRepository store.ResourceRepository, moduleRepository store.ModuleRepository) *Service {
+	return &Service{
+		resourceRepository: resourceRepository,
+		moduleRepository:   moduleRepository,
+	}
+}
+
+func (s *Service) Sync(ctx context.Context, urn string) error {
```

Comment:
Instead of relying on `urn` lets pass the complete `r *Resource` and return resource type, since urn seems like a very low level abstraction to depend on from POV of module service.

Reply:
Sounds good. Let me refactor this.